### PR TITLE
chore: push eol notifications for windows 8 and 8.1

### PR DESCRIPTION
**Pull request by deepak1556**
Refs https://github.com/microsoft/vscode/issues/177676

In preparation for Electron >=23, branch-off from https://github.com/microsoft/vscode/pull/175818

### CODE REVIEW

**Review by isidorn — DISMISSED** [review]

File: src/vs/workbench/electron-sandbox/window.ts [hunk 1]
```diff
@@ -697,23 +697,44 @@ export class NativeWindow extends Disposable {
...
+			]);
 
 			// Refs https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-osversioninfoa
-			if (parseInt(version[0]) === 6 && parseInt(version[1]) === 1) {
-				const message = localize('windows 7 eol', "{0} on Windows 7 will no longer receive any further updates.", this.productService.nameLong);
+			if (eolReleases.get(majorVersion)?.has(minorVersion)) {
+				const message = localize('windowseolmessage', "{0} on {1} will soon stop receiving updates. Consider upgrading your windows version.", this.productService.nameLong, eolReleases.get(majorVersion)?.get(minorVersion));
+				const actions = [{
+					label: localize('windowseolBannerLearnMore', "Learn More"),
+					href: 'https://aka.ms/vscode-faq-old-windows'
```
Comment (deepak1556) [inline src/vs/workbench/electron-sandbox/window.ts:718]: @isidorn I was thinking of unifying the Windows Eol section in FAQ similar to what we did for macOS, do you think it is a good idea ?

File: src/vs/workbench/electron-sandbox/window.ts [hunk 2]
```diff
@@ -697,23 +697,44 @@ export class NativeWindow extends Disposable {
...
 			const version = this.environmentService.os.release.split('.');
+			const majorVersion = version[0];
+			const minorVersion = version[1];
+			const eolReleases = new Map<string, Map<string, string>>([
+				['6', new Map<string, string>([
+					['1', 'Windows 7 / Windows Server 2008 R2'],
+					['2', 'Windows 8 / Windows Server 2012'],
+					['3', 'Windows 8.1 / Windows Server 2012 R2'],
+				])],
+			]);
```
Comment (joaomoreno) [inline src/vs/workbench/electron-sandbox/window.ts:711]: Should this be extracted outside of this method?

File: src/vs/workbench/electron-sandbox/window.ts [hunk 3]
```diff
@@ -697,23 +697,44 @@ export class NativeWindow extends Disposable {
...
+					href: 'https://aka.ms/vscode-faq-old-windows'
+				}];
+
+				this.bannerService.show({
+					id: 'windowseol.banner',
+					message,
+					ariaLabel: localize('windowseolarialabel', "{0}. Use navigation keys to access banner actions.", message),
+					actions,
+					icon: Codicon.warning
+				});
```
Comment (joaomoreno) [inline src/vs/workbench/electron-sandbox/window.ts:727]: Nice 👌

File: src/vs/workbench/electron-sandbox/window.ts (hunk 2 above)
Comment (deepak1556) [reply src/vs/workbench/electron-sandbox/window.ts:711]: Version mappings are different between Windows and MacOS, so keeping them separate is better for simplicity.

**Review by isidorn — APPROVED** [review]

**Comment by isidorn** [comment]
Sorry that my review triggered an auto merge!

File: src/vs/workbench/electron-sandbox/window.ts (hunk 3 above)
Comment (lorand-horvath) [reply src/vs/workbench/electron-sandbox/window.ts:727]: @deepak1556 Please add a neverShowAgain option to the windowseol bannerService in line 721. It's annoying to have to close it every time I start Insiders. I know I'm running an unsupported windows and I don't really need a reminder every time. I want to be able to dismiss it forever. Is that possible to do?

Comment (ajsedwards) [reply src/vs/workbench/electron-sandbox/window.ts:727]: @deepak1556 I second @lorand-horvath request, we do not need a nag every time a project is opened!

**Pull request merged**